### DISCUSSION TITLE
1.0-dev

### DISCUSSION THREAD

**Discussion by brentscheffler** [TomorrowIdeas/plaid-sdk-php#27] [Show and tell]
I am close to wrapping up and publishing the 1.0 release of plaid-sdk-php. I have a couple things I wanted to share with everyone in preparation and get some last minute feedback. Please take a look at the **1.0-dev** branch for more context.

* **Constructor method signature has changed.** Plaid has deprecated their **public_key** support and in response, the constructor will no longer need it.
* **Resource based SDK** With the plethora of endpoints now available, the `Plaid` class has become an unmanageable "God class". I've decided to break apart the endpoints into "resources" accessible from the `Plaid` class. The method names and signature have remained the same to at least minimize the amount of code changes required. For example, the `$plaid->getItems()` method, has been moved to `$plaid->items->getItems()`.
* **We will no longer maintain support for changing the API version.** The library will be hard coded to an API version (the latest supported by Plaid.) If you absolutely *must* be on a previous API version, you will need to lock the SDK version number in your composer file to the Plaid version you need.
* **Minimum supported PHP version has been bumped to 7.3.** In preparation for the PHP 8.0 release, PHPUnit had to be updated. In doing so, they dropped support for PHP 7.2. The library is still, at the time of this writing, fully compatible with PHP 7.2, we are simply dropping it as an officially supported PHP version.

**Comment by syphen** [comment]
Just wanted to say that I am super appreciative of this library and that I really like the resource based setup now. It aligns more with how the official API works and just makes sense. 👍
